spyderlib/spyder.py

```diff
@@ -2653,6 +2660,61 @@ def start_open_files_server(self):
                 self.sig_open_external_file.emit(fname)
             req.sendall(b' ')
 
+    # ---- Quit and restart
+    def restart(self):
+        """ """
+        # Get start path to use in restart script
+        spyder_start_directory = osp.dirname(osp.abspath(__file__))
+        spyder_folder = osp.split(spyder_start_directory)[0]
+        restart_script = osp.join(spyder_start_directory, 'spyder_restart.py')

```

Comment:
We have a function in `baseconfig` called `get_module_path` that would better here :-)